WordPress/build.gradle

```diff
@@ -13,6 +13,57 @@ plugins {
     id "com.google.dagger.hilt.android"
     id "org.jetbrains.kotlinx.kover"
     id "com.google.devtools.ksp"
+    id "com.osacky.fladle"
+}
+
+fladle {
+    testTargets = [
+            "notPackage org.wordpress.android.ui.screenshots"
+    ]
+    configs {
+        wordpress {
+            def app = "wordpress"
+            debugApk.set(pathForVariant(app, "debugApk"))
+            instrumentationApk.set(pathForVariant(app, "instrumentationApk"))
+            smartFlankGcsPath.set(pathForSmartFlankGcsPath(app))
+            testTargets.set(testTargets.get() + [
+                    "notClass org.wordpress.android.e2e.StatsTests",
+                    "notClass org.wordpress.android.e2e.StatsGranularTabsTest",
+            ]
+            )
+        }
+        jetpack {
+            def app = "jetpack"
+            debugApk.set(pathForVariant(app, "debugApk"))
+            instrumentationApk.set(pathForVariant(app, "instrumentationApk"))
+            smartFlankGcsPath.set(pathForSmartFlankGcsPath(app))
+        }
+    }
+    serviceAccountCredentials = new File(System.getProperty("user.home"), ".configure/wordpress-android/secrets/firebase.secrets.json")
+    devices = [
+            ["model": "Pixel2.arm", "version": "30"]
+    ]
+
+    localResultsDir = "$rootDir/build/instrumented-tests"
+
+    maxTestShards = 50
+    shardTime = 120
+}
+
+String pathForVariant(String app, String type) {
```

Comment:
Suggestion (💡): I understand that you tried to hit 2 birds with 1 stone, but consider the below alternative instead, IMHO it seems simpler and less error prone:

```groovy
String appPathForVariant(String app) {
    String path = "outputs/apk/${app}Vanilla/debug"
    return new File(layout.buildDirectory.dir(path).get().asFile, '*.apk').absolutePath
}
String testPathForVariant(String app) {
    String path = "outputs/apk/androidTest/${app}Vanilla/debug"
    return new File(layout.buildDirectory.dir(path).get().asFile, '*.apk').absolutePath
}
```

Then, you could just use those as such:

```groovy
        wordpress {
            def app = "wordpress"
            debugApk.set(appPathForVariant(app))
            instrumentationApk.set(testPathForVariant(app))
        }
        jetpack {
            def app = "jetpack"
            debugApk.set(appPathForVariant(app))
            instrumentationApk.set(testPathForVariant(app))
        }
```

The above, to me, it looks a bit more simplified, wdyt? 🤔

Reply:
Sure, I'll do this - I wasn't sure what will be the most readable, and was going back and forth with this 😅

Reply:
Yea, I was reading all the inverted ifs and more if logic, and couldn't help myself for trying to figure out if we can make this a bit simpler... 🫣 To me, less ifs, less complexity... 😅